[TeaMakesMePee] iP

TeaMakesMePee opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

TeaMakesMePee committed:
renamed, greets and exits

TeaMakesMePee committed:
echo done

TeaMakesMePee committed:
add and list done

TeaMakesMePee committed:
mark/unmark + Task class

TeaMakesMePee committed:
todo event and deadline classes implemented

TeaMakesMePee committed:
A-TextUiTesting

TeaMakesMePee committed:
added exception class with 4 exception types

TeaMakesMePee committed:
delete functionality added

TeaMakesMePee committed:
enums added for exception class

TeaMakesMePee committed:
Task class: Made abstract

TeaMakesMePee committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

TeaMakesMePee committed:
Revert "Merge remote-tracking branch 'upstream/add-gradle-support'"

This reverts commit 4c1a4ffa148081314c6e0c4a6fe3e059f55551ea, reversing
changes made to 8b98bcc476156857905f5137c4a26d6cd7ae0ad5.

TeaMakesMePee committed:
Fix packaging issue

TeaMakesMePee committed:
Reapply "Merge remote-tracking branch 'upstream/add-gradle-support'"

This reverts commit 5caf5d76f04c421f83190e4e377b7907067fe574.

TeaMakesMePee committed:
Add save and load

TeaMakesMePee committed:
Deadline & Event: Add date time

TeaMakesMePee committed:
Merge pull request #1 from TeaMakesMePee/branch-Level-8

Deadline & Event: Add date time

TeaMakesMePee committed:
Add Parser, Data, UI, TaskList Managers

TeaMakesMePee committed:
Add packages

TeaMakesMePee committed:
Add gradle

TeaMakesMePee committed:
Add JUnit Test

TeaMakesMePee committed:
Build jar

TeaMakesMePee committed:
Add comments

TeaMakesMePee committed:
Format coding style

TeaMakesMePee committed:
Add find

TeaMakesMePee committed:
Merge branch 'branch-A-CodingStandard'

TeaMakesMePee committed:
Merge branch 'branch-Level-9'

TeaMakesMePee committed:
Add JavaFX

nicholasohjj commented on src/main/java/tiffy/Tiffy.java:
```diff
@@ -0,0 +1,106 @@
+        }
+    }
+
+    public static void handleRequests(Parser p, String input, TaskManager tm) throws TiffyException {
```
`handleRequests` could be more descriptive – It is responsible for parsing input and executing commands, so something like `processUserCommand `or `executeCommand `might be clearer.

nicholasohjj replied:
Single-character variable names should be avoided unless widely understood (i, n). A more descriptive name like `parser `would be clearer.

nicholasohjj commented on src/main/java/tiffy/Tiffy.java:
```diff
@@ -0,0 +1,106 @@
+
+    public static void markDoneUndone(List<Task> tasks, boolean mark, int index) throws TiffyException {
+        try {
+            Task temp = tasks.get(index - 1);
```
"temp" is too generic. A more descriptive name like `taskToUpdate `or `matchedTasks` would improve readability.

nicholasohjj commented on src/main/java/tiffy/Tiffy.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    public static void handleRequests(Parser p, String input, TaskManager tm) throws TiffyException {
+        String[] partition = p.handleRequests(input);
```
a clearer name like `commandParts `or `parsedInput `would better reflect its purpose.

nicholasohjj reviewed (changes requested):
Good job and keep up the work. some nits to fix

GZTan23 commented on src/main/java/tiffy/Tiffy.java:
```diff
@@ -0,0 +1,106 @@
+            }
+            case "todo" -> {
+                tm.addTask(new Todo(partition[1]));
+            }
```
I believe that the `case` statements should not be indented?

GZTan23 commented on src/main/java/utility/Parser.java:
```diff
@@ -0,0 +1,68 @@
+     * @throws TiffyException If the input is invalid or incomplete.
+     */
+    public String[] handleRequests(String request) throws TiffyException {
+        String[] partition = request.split(" ");
```
I think that 'partition' should be 'partitions' instead since it signifies a collection of Strings

GZTan23 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,97 @@
+ * Represents an abstract Task with a description and a completion status.
+ * This class serves as a base for specific types of tasks.
+ */
+public abstract class Task {
```
Awesome job in making this an abstract class rather than a normal class!

GZTan23 commented on src/main/java/task/Todo.java:
```diff
@@ -0,0 +1,49 @@
+        super(description, isDone);
+    }
+
+    /**
+     * Returns the formatted string representation of the todo task.
+     *
+     * @return Formatted string representing the todo task.
+     */
+    @Override
+    public String getFormattedTask() {
+        return "T|" + this.isDone + "|" + this.description;
+    }
```
Maybe the JavaDoc is not required here? Since you've already included it in the Task class, which this method overrides and the functionality of the method is the same as listed in the Task class.

GZTan23 reviewed (commented):
Great job on the code, seems like you've put in quite the effort! All the best for the rest of the iP!

TeaMakesMePee committed:
Update chatbot UI

The chatbot UI previously used an outdated image for its face, which did not align with the new branding or user experience goals. Additionally, some UI text elements lacked clarity and consistency, leading to potential confusion for users.

An updated chatbot face image has been added to refresh the UI and create a more engaging experience. The UI text has also been refined to improve readability and consistency.

Let's enhance the chatbot’s visual appeal and usability by:

Replacing the existing chatbot image with the new version.
Updating UI text elements to improve clarity and coherence.
These changes improve the user experience, making interactions with the chatbot more intuitive and visually appealing.

TeaMakesMePee committed:
Add assert to key areas

Previously, the code lacked explicit checks for invalid states, which could lead to unexpected runtime errors. Certain methods did not verify preconditions, potentially allowing null values, empty inputs, or out-of-bounds indices to propagate undetected.

To improve code safety and debugging efficiency, assertions have been added to verify internal consistency in critical areas:
- Ensuring that task descriptions are non-null and non-empty in `Task` and `Todo`.
- Validating that task indices are within bounds before accessing lists in `TaskManager` and `Tiffy`.
- Checking that `Parser` correctly splits user input into valid commands before processing.
- Verifying that `handleRequests` receives properly structured input before creating tasks.

These assertions help catch logical errors early without affecting production performance, as they are only active when assertions are enabled (`-ea`).

This update enhances the robustness of task management operations while maintaining clear error handling for invalid user actions.

TeaMakesMePee committed:
Refactor variable names

Some variable names are ambiguous or inconsistent, making the code harder
to read and maintain. The lack of clear naming conventions can lead to
misinterpretations of the code's intent, reducing maintainability and
developer efficiency.

Unclear variable names also make debugging more difficult, as they do not
immediately convey their purpose. This issue affects multiple files,
including TaskManager, Parser, and Tiffy, where variable names
do not fully reflect their roles.

Let's improve readability and maintainability by:

* Renaming variables in TaskManager to clearly indicate task properties.
* Updating Parser variable names for better readability in input handling.
* Refining Tiffy method parameters to distinguish between task actions.

Using meaningful names enhances code clarity, making it easier for future
developers to understand the program’s logic and maintain consistency
across the codebase.

TeaMakesMePee committed:
Merge pull request #2 from TeaMakesMePee/branch-A-Assertions

Add assert to key areas

TeaMakesMePee committed:
Merge branch 'master' into branch-A-CodeQuality

TeaMakesMePee committed:
Merge pull request #3 from TeaMakesMePee/branch-A-CodeQuality

Refactor variable names

TeaMakesMePee committed:
Add contacts feature

Implemented `Contact` class with properties:
* `name` (required)
* `phoneNumber` (required)
* `email` (optional)

Implemented `ContactManager` class to handle:
* Adding contacts with or without an email
* Storing contacts in a list
* Saving and loading contacts from a JSON file (`contacts.json`) using Jackson
* Displaying all stored contacts

Implemented `Main` class with a sample usage of `ContactManager`

Added dependency on `Jackson Databind` for JSON serialization/deserialization

TeaMakesMePee committed:
Refactor request handling and parsing, add contact/task management

Refactored RequestHandler's handleRequest method to reduce its length
to under 30 lines while maintaining clarity. Similarly, optimized the
Parser class by refining the splitRequest method, breaking it down
into smaller methods for better abstraction and maintainability.

Added functionality to:
* List and delete both contacts and tasks.
* Support a new syntax for adding contacts.

These changes improve code readability, maintainability, and extend
application functionality.

TeaMakesMePee committed:
Improve docs, add tests for Parser, Event

Added Javadoc for all classes/methods except GUI and JUnit tests.
Refactored unit tests:
* Added `ParserTest` for `splitRequest()` and `validateCommand()`
* Added `EventTest` for `getFormattedTask()` and `toString()`
* Removed outdated JUnit tests that were not useful
These changes enhance clarity and test coverage.

TeaMakesMePee committed:
Fix Parser validation and request handling logic

Refactored input validation logic for better maintainability:
* Moved `handleDeleteEvent()` validation from `RequestHandler`
  to `Parser` to ensure input errors are caught at the parsing stage
* Updated `splitRequest()` in `Parser` to correctly validate
  input length, preventing missing arguments from reaching
  `RequestHandler`

Improved string splitting logic:
* Modified `handleEvent()` and `handleDeadline()` in `Parser`
  to use `split(" /from | /to ", -1)`, ensuring trailing empty
  elements are preserved for correct argument handling

These changes prevent incorrect input from propagating through
the system and improve overall code robustness.

TeaMakesMePee committed:
Refactor naming and formatting for readability

Refactor naming and formatting for readability

Renamed `eventType` enum to `EventType` in `UiManager.java` to follow
PascalCase naming conventions.

Updated constants:
* Renamed `DATA_FILE_PATH` → `TASKS_JSON_PATH` and
  `CONTACTS_FILE_PATH` → `CONTACTS_JSON_PATH` in `DataManager.java`
  for better clarity

Refactored switch statements:
* Improved indentation and readability in `generateEventFeedback()`
  in `UiManager.java`
* Ensured consistent indentation in `TaskManager.java` for clarity

No functional changes were made. These updates improve readability
and maintainability.

TeaMakesMePee committed:
Fix argument parsing bug in Parser

Resolved an issue in `Parser` where arguments were not correctly
parsed before being passed into the `validateCommand()` function.
This caused incorrect validation behavior for certain commands.

Changes made:
* Ensured `splitRequest()` correctly extracts and formats arguments
  before passing them to `validateCommand()`
* Updated validation logic to correctly handle edge cases where
  arguments were missing or incorrectly formatted

This fix ensures commands are properly validated before execution,
improving input handling reliability.

TeaMakesMePee committed:
Add EventType enum to UiManager for feedback

Introduced a new `EventType` enum in `UiManager` to standardize
feedback generation for different event cases.

Changes made:
* Created `EventType` enum to categorize feedback cases
* Updated `generateEventFeedback()` to use `EventType`, improving
  readability and maintainability
* Refactored related switch-case logic for consistency

This improves UI feedback consistency and makes it easier to extend
event handling in the future.

TeaMakesMePee committed:
Add feedback for contact deletion in UiManager

Updated `UiManager` to generate user feedback when a contact is
deleted, improving interaction clarity.

Changes made:
* Modified `deleteContact()` in `UiManager` to provide a confirmation
  message after successful deletion
* Ensured feedback follows the existing UI format for consistency

This improves user experience by providing clear confirmation when a
contact is removed.

TeaMakesMePee committed:
Add new user profile picture

TeaMakesMePee committed:
Update README and set title

TeaMakesMePee committed:
Update README.md

TeaMakesMePee committed:
Update README.md

TeaMakesMePee committed:
Update UI .png name
